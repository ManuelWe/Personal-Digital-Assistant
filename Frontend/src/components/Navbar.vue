<template>
  <b-navbar>
    <template slot="brand">
      <b-navbar-item tag="div">
        <router-link
          to="/"
        >
          <img
            src="../assets/logo.jpg"
            width="150"
            height="150"
            alt="Logo"
          >
        </router-link>
      </b-navbar-item>
    </template>
    <template slot="end">
      <b-navbar-item tag="div">
        <router-link
          class="button is-white"
          :to="{name: 'dialog', query: {usecase: 'commute'}}"
        >
          <span class="icon">
            <i class="fas fa-train" />
          </span>
          <span>Commute</span>
        </router-link>
      </b-navbar-item>
      <b-navbar-item tag="div">
        <router-link
          class="button is-white"
          :to="{name: 'dialog', query: {usecase: 'travel'}}"
        >
          <span class="icon">
            <i class="fas fa-plane-departure" />
          </span>
          <span>Travel</span>
        </router-link>
      </b-navbar-item>
      <b-navbar-item tag="div">
        <router-link
          class="button is-white"
          :to="{name: 'dialog', query: {usecase: 'restaurant'}}"
        >
          <span class="icon">
            <i class="fas fa-utensils" />
          </span>
          <span>Restaurant</span>
        </router-link>
      </b-navbar-item>
      <b-navbar-item tag="div">
        <router-link
          class="button is-white"
          :to="{name: 'dialog', query: {usecase: 'trainer'}}"
        >
          <span class="icon">
            <i class="fas fa-swimmer" />
          </span>
          <span>Trainer</span>
        </router-link>
      </b-navbar-item>
      <b-navbar-item tag="div">
        <div class="field has-addons">
          <p class="control">
            <router-link
              class="button is-white"
              :class="{'is-outlined': currentRoute !== 'calendar'}"
              to="calendar"
            >
              <span class="icon">
                <i class="far fa-calendar-alt fa-2x" />
              </span>
            </router-link>
          </p>
          <p class="control">
            <router-link
              class="button is-white"
              :class="{'is-outlined': currentRoute !== 'preferences'}"
              to="preferences"
            >
              <span class="icon">
                <i class="fas fa-cog fa-2x" />
              </span>
            </router-link>
          </p>
          <div class="control">
            <router-link
              class="button is-white"
              :class="{'is-outlined': currentRoute !== 'help'}"
              to="help"
            >
              <span class="icon">
                <i class="fas fa-question-circle fa-2x" />
              </span>
            </router-link>
          </div>
        </div>
      </b-navbar-item>
    </template>
  </b-navbar>
</template>

<script>
export default {
  computed: {
    currentRoute() {
      return this.$route.name;
    },
  },
};
</script>

<style scoped>
.navbar-item img {
  max-height: 4rem;
}
.navbar {
  margin: 0 1rem 0 1rem;
}
</style>

<style>
.navbar-burger, .navbar-burger:hover {
  background-color: white;
}
</style>
