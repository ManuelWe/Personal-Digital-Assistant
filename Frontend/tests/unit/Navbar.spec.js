import { createLocalVue, mount } from '@vue/test-utils';
import Buefy from 'buefy';
import VueRouter from 'vue-router';
import Navbar from '@/components/Navbar.vue';

const localVue = createLocalVue();
localVue.use(Buefy);
localVue.use(VueRouter);

const routes = [
  {
    path: '/',
    name: 'landingPage',
    component: Navbar,
  },
];
const router = new VueRouter({
  routes,
});

function factory() {
  return mount(Navbar, {
    localVue,
    router,
  });
}


describe('Navbar.vue', () => {
  describe('Elements are rendered correctly', () => {
    it('Component renders', () => {
      const wrapper = factory();
      expect(wrapper).toMatchSnapshot();
    });

    it('Nav items render', () => {
      const wrapper = factory();
      expect(wrapper.findAll('.navbar-item').length).toBe(6);
    });
  });

  describe('Computed return expected value', () => {
    it('currentRoute works', () => {
      const wrapper = factory();
      expect(wrapper.vm.currentRoute).toBe(routes[0].name);
    });
  });
});
