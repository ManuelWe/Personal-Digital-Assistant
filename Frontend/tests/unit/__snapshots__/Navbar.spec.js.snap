// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue Elements are rendered correctly Component renders 1`] = `
<nav role="navigation" aria-label="main navigation" class="navbar">
  <div class="navbar-brand">
    <div class="navbar-item"><a href="#/" class="router-link-exact-active router-link-active"><img src="../assets/logo.jpg" width="150" height="150" alt="Logo"></a></div><a role="button" aria-label="menu" class="navbar-burger burger"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
  </div>
  <div class="navbar-menu">
    <div class="navbar-start">
      <div class="navbar-item"></div>
    </div>
    <div class="navbar-end">
      <div class="navbar-item"><a href="#/dialog?usecase=commute" class="button is-white is-outlined"><span class="icon"><i class="fas fa-train"></i></span> <span>Commute</span></a></div>
      <div class="navbar-item"><a href="#/dialog?usecase=travel" class="button is-white is-outlined"><span class="icon"><i class="fas fa-plane-departure"></i></span> <span>Travel</span></a></div>
      <div class="navbar-item"><a href="#/dialog?usecase=restaurant" class="button is-white is-outlined"><span class="icon"><i class="fas fa-utensils"></i></span> <span>Restaurant</span></a></div>
      <div class="navbar-item"><a href="#/dialog?usecase=trainer" class="button is-white is-outlined"><span class="icon"><i class="fas fa-swimmer"></i></span> <span>Trainer</span></a></div>
      <div class="navbar-item">
        <div class="field has-addons">
          <p class="control"><a href="#/calendar" class="button is-white is-outlined"><span class="icon"><i class="far fa-calendar-alt fa-2x"></i></span></a></p>
          <p class="control"><a href="#/preferences" class="button is-white is-outlined"><span class="icon"><i class="fas fa-cog fa-2x"></i></span></a></p>
          <div class="control"><a href="#/help" class="button is-white is-outlined"><span class="icon"><i class="fas fa-question-circle fa-2x"></i></span></a></div>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
